### DISCUSSION TITLE
Q-Drawer not appearing on top most layer in the page when using both vertical and horizontal tabs

### DISCUSSION THREAD

**Discussion by L-S-ONG** [quasarframework/quasar#9772] [General - Components / Directives / etc] [answered]
Hi, I am currently using q-drawer for quite a while and recently realized that the q-drawer does not appear to be on the top most layer of the page when using both vertical and horizontal tabs

I have tried to style my header with z-index: 0, but the problem is, buttons on q-header will be unable to click
https://codepen.io/l-s-ong/pen/ExWBgyO
(this is how I would like the q-drawer to present, it overlays the q-header)
![image](https://user-images.githubusercontent.com/56607625/123059726-a36eb580-d43c-11eb-8717-a22076342908.png)

v.s

https://codepen.io/l-s-ong/pen/wvJLGLZ
this is my current way of doing, the q-header is overlapping my right drawer
![image](https://user-images.githubusercontent.com/56607625/123059795-b7b2b280-d43c-11eb-9aaf-368c1422e991.png)
^ What should I do to this to make the right drawer overlaps the entire page?

**Comment by MilosPaunovic** [comment]
Hello @L-S-ONG,

You should take out `QDrawer` element out of the `QContainer` from your Codepen exapmle. Try placing that element right below the left drawer element and check if that is what you are looking to achieve.

**Reply by L-S-ONG** [reply]
It works, however, because for my project, I have already separated each components and the QDrawer has to be within QContainer. Could there be other workaround to solving this issue?

**Reply by MilosPaunovic** [reply]
If you are set on keeping the drawer inside a `QContainer`, I don't think there is any remotely elegant solution.

**Reply by L-S-ONG** [reply]
Thank you for your answer @MilosPaunovic. Based on both @rstoenescu and your answers, I'll attempt taking `QDrawer` out of the container. Much appreciated to your kind and prompt reply

**Comment by rstoenescu** [comment, marked as answer]
Please place QDrawer under QPage or QLayout, but don't place it so deep inside of the DOM. The effect that you are experiencing is actually in accordance with the z-index web specs.
One more reason for not needing to place QDrawer so deep inside of the DOM is that it doesn't have to be side by side with the button that "activates" it. All that matters is the vue scope.

**Reply by L-S-ONG** [reply]
Thanks for the explanation, I'll try not to place my QDrawer so deep inside the DOM.
